qiskit/providers/aer/extensions/initialize_ket.py

```diff
@@ -0,0 +1,263 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Initialize qubit registers to desired arbitrary state specified by dictionary.
+"""
+
+import math
+import numpy as np
+
+from qiskit.exceptions import QiskitError
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit import QuantumRegister
+from qiskit.circuit import Instruction
+from qiskit.circuit.library.standard_gates.x import CXGate
+from qiskit.circuit.library.standard_gates.ry import RYGate
+from qiskit.circuit.library.standard_gates.rz import RZGate
+from qiskit.circuit.reset import Reset
+
+_EPS = 1e-10  # global variable used to chop very small numbers to zero
+
+
+class InitializeKet(Instruction):
```

Comment:
This function should be added to Terra. If it currently works with unrolling to reset + basis gates like regular initialization we could try and add it to the 0.15 release which is coming out any day now.

Reply:
It needs some sort of ket2vec to mimic what the regular initialize does. Do we have one in Python? If not, I'll write one myself and transfer to Terra.